packages/mdc-icon-button/index.ts

```diff
@@ -22,34 +22,27 @@
  */
 
 import MDCComponent from '@material/base/component';
-import MDCIconButtonToggleFoundation from './foundation';
 import {MDCRipple} from '@material/ripple/index';
+import MDCIconButtonToggleFoundation from './foundation';
 
-/**
- * @extends {MDCComponent<!MDCIconButtonToggleFoundation>}
- */
-class MDCIconButtonToggle extends MDCComponent {
-  static attachTo(root) {
+class MDCIconButtonToggle extends MDCComponent<MDCIconButtonToggleFoundation> {
+  static attachTo(root: Element) {
     return new MDCIconButtonToggle(root);
   }
 
-  constructor(...args) {
-    super(...args);
+  private ripple_: MDCRipple;
+  private handleClick_: EventListenerOrEventListenerObject;
 
-    /** @private {!MDCRipple} */
-    this.ripple_ = this.initRipple_();
-    /** @private {!Function} */
-    this.handleClick_;
-  }
+  constructor(
+    root: Element,
+    foundation?: MDCIconButtonToggleFoundation,
+    // tslint:disable-next-line:no-any a component can pass in anything it needs to the constructor
+    ...args: any[]
```

Comment:
I don't think we event need `...args`. Its only used in `mdc-base/component`'s `initialize` method. And there is not `initialize method.

Reply:
We probably shouldn't override the constructor. Instead, initialize the properties in `initialSyncWithDOM()` and mark them with `!`:

```js
  private ripple_!: MDCRipple;
  private handleClick_!: EventListenerOrEventListenerObject;
```